tests/test_api_requests.py

```diff
@@ -137,6 +139,124 @@ async def test_http_response_headers_request(
+        ({}, {"ZYTE_API_ON_ALL_REQUESTS": False}, {}, False),
+        ({"zyte_api": {}}, {"ZYTE_API_ON_ALL_REQUESTS": False}, {}, False),
+        ({"zyte_api": True}, {"ZYTE_API_ON_ALL_REQUESTS": False}, {}, False),
+        ({"zyte_api": False}, {"ZYTE_API_ON_ALL_REQUESTS": False}, {}, False),
+        (
+            {},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": False,
+            },
+            {},
+            False,
+        ),
+        (
+            {"zyte_api": False},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": False,
+            },
+            {},
+            False,
+        ),
+        (
+            {"zyte_api": None},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": False,
+            },
+            {},
+            False,
+        ),
+        (
+            {"zyte_api": {}},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": False,
+            },
+            {"browserHtml": True, "geolocation": "CA"},
+            True,
+        ),
+        (
+            {"zyte_api": True},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": False,
+            },
+            {"browserHtml": True, "geolocation": "CA"},
+            True,
+        ),
+        (
+            {"zyte_api": {"javascript": True, "geolocation": "US"}},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": False,
+            },
+            {"browserHtml": True, "geolocation": "US", "javascript": True},
+            True,
+        ),
+        # ZYTE_API_ON_ALL_REQUESTS=True
+        ({}, {"ZYTE_API_ON_ALL_REQUESTS": True}, {}, False),
+        ({"zyte_api": {}}, {"ZYTE_API_ON_ALL_REQUESTS": True}, {}, False),
+        ({"zyte_api": True}, {"ZYTE_API_ON_ALL_REQUESTS": True}, {}, False),
+        ({"zyte_api": False}, {"ZYTE_API_ON_ALL_REQUESTS": True}, {}, False),
+        (
+            {},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": True,
+            },
+            {"browserHtml": True, "geolocation": "CA"},
+            True,
+        ),
+        (
+            {"zyte_api": False},
+            {
+                "ZYTE_API_DEFAULT_PARAMS": {"browserHtml": True, "geolocation": "CA"},
+                "ZYTE_API_ON_ALL_REQUESTS": True,
+            },
+            {},
+            False,
+        ),
+        (
+            {"zyte_api": None},
```

Comment:
In the current README docs, it's described as:

> Set the ``ZYTE_API_ON_ALL_REQUESTS`` setting to ``True`` to make all requests
go through Zyte Data API unless the ``zyte_api`` key in Request.meta_ is set to
``False``. ``ZYTE_API_ON_ALL_REQUESTS`` is ``False`` by default.

I'm concerned that the docs isn't portraying the correct message since a value of `None` doesn't go through ZDA while the test case below with `{}` goes through.

Should we disable the ZDA calls _**strictly**_ if the value is `False` and not _false-like_? 

Alternatively, we could also mention that both `False` and `None` values are supported in the README.

Reply:
`None` works, just like `0`, but these changes include deprecating them, and that is why they are not even mentioned in the documentation.

Users should use `False` or a `dict`. Anything else is deprecated. For backward compatibility `None` and `0` will work as before, but they should not be used in any new code.

Reply:
Gotcha. I totally forgot about https://github.com/scrapy-plugins/scrapy-zyte-api/pull/40/files#diff-ed75145a6134504c5e20e23c0958e4f4929703abac118e0968f055d51a5fde7bR80-R84